src/components/Choropleth.vue

```diff
@@ -55,50 +52,84 @@ const backgroundLayer = {
 
 const map = ref<typeof LMap | null>(null);
 const featureRefs = ref<(typeof LGeoJson)[]>([]);
+const bounds: Ref<LatLngBounds | null> = ref(null);
 
 const router = useRouter();
-const { selectedFeatures, selectedIndicators, loading, selectedIndicator, selectedCountryId, appConfig } =
-    storeToRefs(useAppStore());
+const {
+    selectedFeatures,
+    selectedIndicators,
+    loading,
+    selectedIndicator,
+    selectedCountryId,
+    appConfig,
+    countryBoundingBoxes,
+    waitingForMapBounds
+} = storeToRefs(useAppStore());
+
+const showSpinner = computed(() => waitingForMapBounds.value || loading.value);
+useLoadingSpinner(map, showSpinner);
 
-useLoadingSpinner(map, loading);
 const { getColour } = useColourScale(selectedIndicators);
 
-const getFeatureId = (feature: Feature) => feature.properties![FEATURE_ID_PROP];
-const getFeatureName = (feature: Feature) => feature.properties![FEATURE_NAME_PROP];
+const featureProps = computed(() => appConfig.value?.geoJsonFeatureProperties);
```

Comment:
For readability, I would vote for using the full term featureProperties to avoid conflation with Vue component props.

Reply:
Yep, fair enough, will update that.